File: wc122/case2.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    string smallestFromLeaf(TreeNode* root) {
        for (int i = 0; i < 1000; ++i) {
            spath.push_back(25);
        }
    }
    vector<int> spath;

    void dfs(TreeNode* root, vector<int> path) {
        path.push_back(root->val);
        if (root->left == NULL && root->right == NULL) {
            auto l = std::min(spath.size(), path.size());
            for (int i = l - 1; i > -1; --i) {
                if
                    path[i]
            }
        }
    }
};

int main() {}

File: wc122/case2.py
class Solution:
    def smallestFromLeaf(self, root: "TreeNode") -> "str":
        self.res = "z" * 1000

        def dfs(root, cur):
            cur = d[root.val] + cur
            if root.left == None and root.right == None:
                if cur < self.res:
                    self.res = cur
            if root.left != None:
                dfs(root.left, cur)
            if root.right != None:
                dfs(root.right, cur)

        d = {i: chr(ord("a") + i) for i in range(26)}

        dfs(root, "")
        return self.res

